features/volgindicaties.feature

```diff
@@ -0,0 +1,154 @@
+# language: nl
+
+Functionaliteit: Als gebruiker wil ik wijzigingen kunnen opvragen van zowel binnengemeentelijke als buitengemeentelijke personen
```

Comment:
Nu ik de scenario's lees, denk ik dat de titel van deze functionaliteit niet correct is. Volgens mij wordt in deze feature bestand beschreven wanneer en welke berichten er naar het GBA-V moet worden gestuurd

Reply:
ik heb overwogen het "volgindicaties-buitengemeentelijk.feature" te noemen, maar dat dekt ook niet helemaal de lading. Het gaat ook over volgindicaties binnengemeentelijk en over het wisselen van binnen- naar buitengemeentelijk of andersom.
Hoe zou jij deze feature noemen?

Reply:
Ik zou het splitsen in meerdere feature bestanden. Minimaal één voor het beschrijven van het gedrag van de Update API en één voor het beschrijven van de communicatie met het GBA-V. Voordeel hiervan is bijv. als het gedrag van het GBA-V wijzigt, dat je alleen het feature bestand mbt communicatie met het GBA-V hoeft aan te passen.

Reply:
Ik denk dat `mappen van Update API volgindicaties naar GBA-V volgindicaties` een betere naam voor deze functionaliteit zou zijn. En als beschrijving zou ik zeggen:

Als Gemeente
Wil ik dat volgindicaties geplaatst/beëindigd met de Update API correct wordt gemapt naar GBA-V volgindicaties
Zodat ik alleen buitengemeentelijke personen bij het GBA-V volg
En alleen als er lopende/actieve Update API volgindicaties zijn voor deze personen

Reply:
Ik zou gebruik maken van de `Rule` keyword om scenario's te groeperen. Nu worden met verschillende scenario's beschreven dat als een persoon geen actieve volgindaties heeft dat een Ap01 bericht naar het GBA-V moet worden gestuurd. Bijv.

**Rule**: stuur Ap01 als een buitengemeentelijk persoon geen actieve volgindicaties heeft

  **Voorbeeld**: Een gebruiker zet als eerste een volgindicatie op een buitengemeentelijk persoon
     # Hier komt dan de stappen van deze scenario

  **Voorbeeld**: Een gebruiker zet een volgindicatie op een persoon die niet meer gevolgd wordt door een andere gebruiker
     # Hier komt dan de stappen van deze scenario

  **Voorbeeld**: Een gebruiker heropent een beëindigde volgindicatie
     # Hier komt dan de stappen van deze scenario

  **Voorbeeld**: Een gebruiker heropent een beëindigde volgindicatie die eerder door een andere gebruiker gevolgd werd
     # Hier komt dan de stappen van deze scenario
     # De titel van deze scenario klopt niet. Volgens mij kan je je volgindicatie alleen heropenen als je eerder één hebt geplaatst en dat is in deze scenario niet het geval

Andere rules die ik zou opnemen

**Rule**: stuur geen Ap01 als een buitengemeentelijk persoon actieve volgindicaties heeft
**Rule**: stuur geen Ap01 voor een binnengemeentelijk persoon
**Rule**: stuur een Av01 als een actief gevolgd persoon niet meer wordt gevolgd

Hiermee wordt het (denk ik) duidelijker welke scenario's horen bij welke hoofdregel.